repository.go

```diff
@@ -1,128 +1,279 @@
 package function
 
 import (
-	"errors"
-	"io/ioutil"
+	"io"
+	"os"
 	"path/filepath"
 	"strings"
 
 	"github.com/go-git/go-git/v5"
 	"github.com/markbates/pkger"
+	"github.com/markbates/pkger/pkging"
+	"gopkg.in/yaml.v2"
 )
 
 // Path to builtin repositories.
 // note: this constant must be defined in the same file in which it is used due
 // to pkger performing static analysis on source files separately.
 const builtinRepositories = "/templates"
 
-// Repository
+const ManifestYaml = "manifest.yaml"
+const RuntimeYaml = "runtime.yaml"
+
+// Repository represents a collection of one or more runtimes, each with one or more templates
 type Repository struct {
-	Name      string
-	URL       string // (empty if not a git repo with an upstream URL)
-	Templates []Template
-	Runtimes  []string
+	Name            string `yaml:"name"`
+	Version         string `yaml:"version,omitempty"`
+	URL             string `yaml:"url,omitempty"`
+	Runtimes        []Runtime
+	HealthEndpoints HealthEndpoints
+}
+
+// FunctionTemplates is the collection of templates for a Runtime
+type FunctionTemplates []FunctionTemplate
+
+// FunctionTemplate is the name and path to a Runtime template
+type FunctionTemplate struct {
+	Name string `yaml:"name,omitempty"`
+	Path string `yaml:"path"`
+}
+
+// HealthEndpoints specify the liveness and readiness endpoints for a Runtime
+type HealthEndpoints struct {
+	Liveness  string `yaml:"liveness,omitempty"`
+	Readiness string `yaml:"readiness,omitempty"`
+}
+
+// Runtime is a path to a directory containing one or more function templates
+type Runtime struct {
+	Path       string            `yaml:"path"`
+	Name       string            `yaml:"name,omitempty"`
+	Buildpacks []string          `yaml:"buildpacks,omitempty"`
+	Builders   map[string]string `yaml:"builders,omitempty"`
+	Templates  FunctionTemplates
+	HealthEndpoints
```

Comment:
This would be needed to be reverted then: https://github.com/knative-sandbox/kn-plugin-func/pull/558/files#diff-3bc17fd6241d546478336fa19a33f7e7f4c232421ce4e8511512c00fa617860dL163-R167